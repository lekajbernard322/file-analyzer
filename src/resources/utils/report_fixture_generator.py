from os import listdir
from os.path import isfile, join
import json
from datetime import datetime, timezone
from bson.objectid import ObjectId

INPUT_DIR = r'../report_data/'
OUTPUT_PATH = r'../../backend/app/report/fixtures/report_fixture.json'

file_paths = [(join(INPUT_DIR, file_name), file_name)
              for file_name in listdir(INPUT_DIR)
              if isfile(join(INPUT_DIR, file_name))]


def prepare_output_data(i, file_name, content):
    return {
        'model': 'report.report',
        'pk': str(ObjectId()),
        'fields': {
            'name': file_name,
            'content': content,
            'created': datetime.now(tz=timezone.utc).isoformat()
        }
    }


data = []
for index, (input_file_path, input_file_name) in enumerate(file_paths):
    print(f'[*] Processing file: {input_file_path}')

    with open(input_file_path, 'r') as input_file:
        json_content = json.load(input_file)

        output_data = prepare_output_data(index, input_file_name, json_content)

        data.append(output_data)

print(f'[*] Dumping to fixture file: {OUTPUT_PATH}')
with open(OUTPUT_PATH, 'w', encoding='utf-8') as output_file:
    json.dump(data, output_file, indent=4)

print(f'[\u2714] Done')
