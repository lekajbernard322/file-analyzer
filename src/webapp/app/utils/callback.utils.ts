export const createOnPressEnter = (callback) => ({key}) => key === 'Enter' && callback && callback()
