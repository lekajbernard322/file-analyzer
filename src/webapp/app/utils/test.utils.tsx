import thunk from "redux-thunk";
import {createMemoryHistory} from "history";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import React from "react";
import configureStore from "redux-mock-store";

export const renderWithContext = (additionalState, ui) => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(additionalState);
    const history = createMemoryHistory();

    const Wrapper = ({children}) => (
        <Provider store={store}>
            <Router history={history}>
                {children}
            </Router>
        </Provider>);

    return {rendered: render(ui, {wrapper: Wrapper}), store, history};
};
