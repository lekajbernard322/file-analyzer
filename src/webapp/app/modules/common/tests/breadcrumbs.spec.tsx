import React from "react";
import {fireEvent} from "@testing-library/react";
import Breadcrumbs from "../breadcrumbs";
import {renderWithContext} from "../../../utils/test.utils";

const mockBreadcrumbs = [{label: 'Reports', to: '/reports'}]

describe("Breadcrumbs", () => {
    const renderComponent = () => {
        return renderWithContext(null, <Breadcrumbs config={mockBreadcrumbs}/>)
    }

    it("should render correctly", () => {
        const {rendered: {container}} = renderComponent();

        expect(container).toMatchSnapshot();
    });

    it("should route to / when clicked on Report breadcrumb", () => {
        const {rendered: {queryByText}, history} = renderComponent();

        const breadcrumb = queryByText(/Reports/);

        fireEvent.click(breadcrumb);

        expect(history.location.pathname).toEqual("/reports");
    })
});
