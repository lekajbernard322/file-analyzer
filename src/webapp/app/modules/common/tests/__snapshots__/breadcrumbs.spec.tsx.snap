// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render correctly 1`] = `
<div>
  <div
    class="breadcrumb"
  >
    <a
      href="/reports"
    >
      Reports
      /
    </a>
  </div>
</div>
`;
